fluentd/out_syslog.rb

```diff
@@ -81,17 +97,31 @@ def emit(tag, es, chain)
           time = Time.now
         end
         @packet.time = time
-        @packet.tag = if @tag_key
-                        begin
-                          record[@tag_key][0..31].gsub(/[\[\]\s]/,'') # tag is trimmed to 32 chars for syslog_protocol gem compatibility
-                        rescue
-                          tag[0..31] # tag is trimmed to 32 chars for syslog_protocol gem compatibility
-                        end
-                      else
-                          tag[0..31] # tag is trimmed to 32 chars for syslog_protocol gem compatibility
-                      end
+        if @tag_key.any?
+          @tag_key.each { |tkey|
+            mytag = record
+            tkey.split('.').each { |p|
```

Comment:
for example - if record contains `{'a' => 'b'}` and the user specifies the key like `a.b` then you will see an error like this:
```
NoMethodError: undefined method `key?' for "a":String
```
That is, the user specified a key that does not exist.
I guess in this case, we should use the default value - `tag[0..31]`.
This could be a misconfiguration, but we don't do any checking for any of the other key parameters that the user can specify. For example, below, if the user misspells the `@payload_key` then `packet.content` will always be `nil`.

But we don't want to emit a big error every time - what if the user knows that not every record will have that field?  Then we are spamming the fluentd error log with many, many messages.

I'd like to say that we have some sort of flag e.g. `@report_missing_key` that defaults to `true`, and we emit an error message when the record key is not found if this is set to `true`.  And, that value can be set via an env. var. e.g. `report_missing_key '#{REMOTE_SYSLOG_REPORT_MISSING_KEY}'`.

If I were an admin, I would like to know if I misconfigured something, but I would also like to have a knob with which to suppress log messages.

@portante - too many knobs?  or too much spam?

Reply:
in the interest of moving this along - don't add the config parameter - just catch the exception and break - treat it just like `! mytag.is_a? String`

Reply:
I added this debug print for the purpose.
https://github.com/nhosoi/origin-aggregated-logging/blob/bz1515715/fluentd/out_syslog_buffered.rb#L71

The following is a snippet of the fluentd log from my test run, when I ssue the message with the warning level.  This is too noisy and not so useful... :(
[2018-03-30T00:25:44.216+0000] 2018-03-30 00:25:29 +0000 [warn]: out:syslog_buffered: u from systemd.u.SYSLOG_IDENTIFIER in tag_key ["ident", "systemd.u.SYSLOG_IDENTIFIER", "err.local1"] is not a key of record.
[2018-03-30T00:25:44.216+0000] 2018-03-30 00:25:29 +0000 [warn]: out:syslog_buffered: err from err.local1 in tag_key ["ident", "systemd.u.SYSLOG_IDENTIFIER", "err.local1"] is not a key of record.

Reply:
ok, but please change the log level to debug instead of warn - we don't want this logged by default

Reply:
hmm - I see below that it is logged at `debug` level - ok